app/models/account.rb

```diff
@@ -40,34 +49,38 @@ def create(id, owner_id = nil)
     end
 
     def list
-      accounts = []
+      accounts = Set.new
       Slosilo.each do |k,v|
         accounts << k
       end
       accounts.map do |account|
-        account =~ /\Aauthn:(.+)\z/
+        account =~ /\Aauthn:([^:]+)(?::[^:]+)?\z/
         $1
       end.delete_if do |account|
         account == "!"
       end
     end
   end
 
-  def token_key
-    Slosilo["authn:#{id}"]
+  def token_key_host
```

Comment:
What do you think about combining these two methods with the above slosilo id methods to make them a bit more generic, something like:

```ruby
def token_key(role)
  Slosilo[token_id(role)]
end

def token_id(role)
  "authn:#{account}:#{role}"
end
```